src/index.ts

```diff
@@ -1,3 +1,4 @@
+import {} from '@fastify/swagger';
```

Comment:
ah, this part does the type augmentation?
`@fastify/swagger` will have to become runtime dependency for this to work, I assume.

Reply:
Yes, this part is needed for types.

Reply:
This will fail compilation if fastify/swagger is not already a dependency of the project

not sure if it will also fail at runtime for JS projects, probably we can somehow mark this as types-only import, TS had something for that

Reply:
The compiled JS will have omitted this line. In general TS is pretty smart about omitting imports that aren't used.

If you want a type-only import the syntax is:

```ts
// For named types
import type { Foo } from 'bar'

// For typed default exports
import type Buzz from 'bang'

// You cannot mix these two styles however
```

But there's no way to do that for an empty set of named imports.

Reply:
@rupert-mckay are you ok with fastify-swagger staying devDependency, if it's only mandatory for TS?
we probably need a note in readme, though, and it's a semver major

Reply:
I don't see any import in the JS file, I assume it compiles like type imports and omits.

Reply:
yeah, but for ts projects it's now a mandatory dependency that is not installed by default

Reply:
I don't think I see the problem here. I use that import just to see the possible schema fields while developing the package which is devDeps should be fine.

Reply:
if someone depends on fastify-type-provider-zod and uses TS, but doesn't have fastify-swagger as a dependency, their CI will break.

Reply:
Yeah something isn't right about this.

> to see the possible schema fields while developing

Where exactly do you mean? Which variables in this file are affected by the presence of this import?

Reply:
I don't think it is a problem, but if you think it's not right. I can remove it and we still can use the `...rest` but not `hide`, therefore there will not be a short circuit for hiding. Everything will work as same.

Reply:
it should be fine if we either explicitly mention dependency on fastify/swagger in docs, or make it a runtime dependency

Reply:
Aha I understand what is going on now. 

Rather than use this import at all, you can just grab the bit of the interface extension that it does internally.

- Remove the import
- Add the following code after the imports, but before any other code:

```ts
  interface FastifySchema {
    hide?: boolean;
    deprecated?: boolean;
    tags?: string[];
    description?: string;
    summary?: string;
    consumes?: string[];
    produces?: string[];
    externalDocs?: {
        [index: string]: any;
        description?: string;
        url: string;
    } | {
        description?: string;
        url: string;
    };
    security?: Array<{ [securityLabel: string]: string[] }>;
    /**
     * OpenAPI operation unique identifier
     */
    operationId?: string;
  }
```

Reply:
Alternatively just take only the bits of this interface that you need. Sounds like you wanted `hide`? Feel free to remove anything that isn't directly relevant to this project. The above is extending the existing FastifySchema type, but you only need to extend it as much as necessary for your usage.

Reply:
@turkerdev so how would you like to proceed with this?

Reply:
I think, we can just use `...rest` option and remove the short circuit.

Reply:
@turkerdev that would be less convenient to use, though. maybe we could add manual augment just for hide, like Rupert suggested? or just a ts-ignore

Reply:
I rather choose manual augment over ts-ignore.